src/uct/ib/ud/base/ud_iface.c

```diff
@@ -274,7 +274,8 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, const uct_ud_iface_config_t *config
     }
 
     self->config.max_inline = qp_init_attr.cap.max_inline_data;
-    uct_ib_iface_set_max_iov(&self->super, qp_init_attr.cap.max_send_sge);
+    uct_ib_iface_set_max_iov(&self->super, qp_init_attr.cap.max_send_sge,
+                             UCT_IB_MAX_IOV);
```

Comment:
should we move it to verbs code only?

Reply:
will also need to revise/separate am_zcopy handling

Reply:
yep

Reply:
btw, only AM is relevant for UD and it is had coded to 3 for mlx5. 
So probably some asserts would be enough